platforms/documentation/docs/src/docs/userguide/platforms/jvm/toolchains.adoc

```diff
@@ -128,6 +145,139 @@ A specification without a language version, in most cases, would be treated as a
 
 Usage of _invalid_ instances of `JavaToolchainSpec` results in a build error since Gradle 8.0.
 
+[[sec:source-target-toolchain]]
+=== 2. Source and Target compatibility
+
+Setting `sourceCompatibility` and `targetCompatibility` tells the Java compiler to produce bytecode compatible with a specific Java version but does *not* enforce which JDK Gradle itself runs with:
+
+[source,kotlin]
+----
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+----
+
+This does *not* guarantee the correct JDK is used and may cause issues when APIs have been backported to older Java versions.
+
+You should only use this method in cases where you need backward compatibility **but cannot use toolchains**.
+
+[[sec:flag-toolchain]]
+=== 3. The `--release` flag
```

Comment:
💭  By the time I scrolled until here, I am lost. I don't know what "list" this third item refers to.
The first toolchains sub-section is very long and the small difference in the heading size does not really help recovering the context. I could have consulted the page structure (rendered on the right), but it only shows top-level stuff, so it's not helpful to orient myself.

We don't have to do anything about it in this PR, but it's a general concern worth thinking over.